### PR TITLE
v2.4.14 upgrade for openharmony

### PR DESCRIPTION
**Pull request by zhaoyang57**
1.Upgrade openharmony template to API11.
2.Add support for custom fonts.
3.Add game pause processing when slided to the multitask list. 
4.Add screen safe area adaptation.
5.Multi-fingers touch bug fix.

### CODE REVIEW

File: templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js [hunk 1]
```diff
@@ -167,5 +171,46 @@ globalThis.vibrate = function (duration) {
 }
 
 globalThis.terminateProcess = function () {
-    this.pro.exit(0);
+    pro.exit(0);
```
Comment (qiuguohua) [inline templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js:174]: When do we need to exit a process?

Comment (zhaoyang57) [reply templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js:174]: When to quit the game, cc.game.end()  will call it.

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 2]
```diff
@@ -58,23 +58,21 @@ void dispatchTouchEventCB(OH_NativeXComponent* component, void* window) {
...
-        ev->type = cocos2d::TouchEvent::Type::CANCELLED;
-    }
     for(int i = 0; i < touchEvent.numPoints; ++i) {
         cocos2d::TouchInfo touchInfo;
         touchInfo.index = touchEvent.touchPoints[i].id;
         touchInfo.x = touchEvent.touchPoints[i].x;
         touchInfo.y = touchEvent.touchPoints[i].y;
-        if (touchEvent.id == touchInfo.index) {
-            ev->touches.push_back(touchInfo);
+        if (touchEvent.touchPoints[i].type == OH_NATIVEXCOMPONENT_DOWN) {
```
Comment (qiuguohua) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp]: Do a transformation using a function?

File: templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js [hunk 3]
```diff
@@ -167,5 +171,46 @@ globalThis.vibrate = function (duration) {
...
+    display.getDefaultDisplaySync().getCutoutInfo().then((data) => {
+        if (data.boundingRects.length == 0) {
+            return;
+        }
+
+        cutout.left = data.boundingRects[0].left;
+        cutout.top = data.boundingRects[0].top;
+        cutout.width = data.boundingRects[0].width;
+        cutout.height = data.boundingRects[0].height;
+    }).catch((err) => {
```
Comment (qiuguohua) [inline templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js:187]: Do we need to add a print error log?

File: templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js [hunk 4]
```diff
@@ -167,5 +171,46 @@ globalThis.vibrate = function (duration) {
...
+    if(!cutout.height) {
+        return 0;
+    }
+
+    let disPlayHeight = display.getDefaultDisplaySync().height;
+    let orientation = globalThis.getDeviceOrientation();
+    if (orientation == display.Orientation.PORTRAIT) {
+        return cutout.top + cutout.height;
+    } else if(orientation == display.Orientation.PORTRAIT_INVERTED) {
+        return disPlayHeight - cutout.top;
```
Comment (qiuguohua) [inline templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js]:
```suggestion
       let disPlayHeight = display.getDefaultDisplaySync().height;
        return disPlayHeight - cutout.top;
```

File: templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js [hunk 5]
```diff
@@ -167,5 +171,46 @@ globalThis.vibrate = function (duration) {
...
+    }
+    return cutout.left + cutout.width;
+}
+
+globalThis.getCutoutHeight = function () {
+    if(!cutout.height) {
+        return 0;
+    }
+
+    let disPlayHeight = display.getDefaultDisplaySync().height;
```
Comment (qiuguohua) [inline templates/js-template-default/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/cocos/sys-ability-polyfill.js]: displayHeight？

**Review by qiuguohua — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**